### PR TITLE
Adding practice questions for topic09.

### PR DESCRIPTION
**Pull request by ikaroruan**

### CODE REVIEW

File: topic09/topic-8.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+author: Ikaro Costa
+---
+
+Practice Questions
+==================
+
+Question 1
+----------
+
+In COMP 1002, you have successfully built your personal webpage. We are now
```
Comment (fbristow) [inline topic09/topic-8.md:12]: Please add a link to the topic where we did this.

File: topic09/topic-8.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Question 2
+----------
+
+Now, we improve the shell script `publish_webpage` from Question 1. 
+
+### Input Error Prevention
+
+Imagine
+that you shared your script with a friend but he did not check your implementation.
```
Comment (fbristow) [inline topic09/topic-8.md:43]: Please do not use gendered pronouns to describe general people in English. Prefer instead to use "they" or "them". In this case, change this sentence to be

> Imagine that you shared your script with a friend but **they** did not...

File: topic09/topic-8.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+The idea here is to prevent that your friend type an incorrect name as the argument
+of `publish_webpage`. For this, before performing any action in the script
+`publish_webpage`, check if the file passed on the command line by the user actually
+exists. If the file do not exist, print to standard output "File not found!" using `echo`
+command. Otherwise, print "File found!" and execute the rest of the script.
```
Comment (fbristow) [inline topic09/topic-8.md:48]: Can you add either a reminder or a link back to the file operators?

File: topic09/topic-8.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+----------
+
+Using a glob to find all the `zip` files in a directory, write a shell script
+called `unzipall` that uses a `for` loop to iterate over all the `zip` files,
+then use `unzip` to extract the files from the compressed format.
+
+::: challenge
+The use of glob can remove the need of loops in many actions on a shell script.
+Globs are meant to make our life easier. Can you write the same `unzipall` command
+but not using a `for` loop?
```
Comment (fbristow) [inline topic09/topic-8.md:60]: This is an excellent challenge!

File: topic09/topic-8.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+`publish_webpage`, check if the file passed on the command line by the user actually
+exists. If the file do not exist, print to standard output "File not found!" using `echo`
+command. Otherwise, print "File found!" and execute the rest of the script.
+
+Question 3
+----------
+
+Using a glob to find all the `zip` files in a directory, write a shell script
+called `unzipall` that uses a `for` loop to iterate over all the `zip` files,
+then use `unzip` to extract the files from the compressed format.
```
Comment (fbristow) [inline topic09/topic-8.md:55]: Can you please prepare something for the practice, either:

1. Prepare a zip file containing a bunch of zip files, or
2. Write a little command that will _create_ a bunch of zip files.

**Review by fbristow — CHANGES_REQUESTED** [review]
This overall looks great, thank you @ikaroruan!

You're overall aligning the assessment with the learning outcomes.

Please make the noted minor changes and this can be merged.

**Comment by ikaroruan** [comment]
Great! I have done the changes and it is now ready to be merged.

**Pull request merged**